src/layman/user/util.py

```diff
@@ -100,3 +105,55 @@ def get_username_suggestions_from_claims(claims):
 def delete_user(username=None):
     authn_redis.delete_user(username)
     delete_whole_user(username)
+
+
+def delete_user_public_publications(username):
+    skipped_publications = []
+    result = get_publication_infos_with_metainfo(reader=username)
+    for key, pubinfo in result['items'].items():
+        workspace, publication_type, publication_name = key
+        is_public = pubinfo.get('_is_public_workspace')
+        access_rights = pubinfo.get('access_rights')
+        geodata_type = pubinfo.get('geodata_type')
+        is_only_rw = (
+            set(access_rights.get('read', [])) == {username}
+            and set(access_rights.get('write', [])) == {username}
+        )
+        publication_name = pubinfo['name']
+
+        if is_public and is_only_rw:
+            if publication_type == LAYER_TYPE:
```

Comment:
You can use `layman.util.delete_workspace_publication` to delete any publication.